### ISSUE TITLE
sbt: fix and/or document usage of akkaGrpcExtraGenerators

### ISSUE THREAD

**Issue by bjaglin** [akka/akka-grpc#1096]
Custom generators declared and compiled as part of the metabuild could be used as `akkaGrpcExtraGenerators` until Akka-gRPC 1.0.0. This is no longer possible as of 1.0.1 (`java.lang.ClassNotFoundException`), since https://github.com/akka/akka-grpc/pull/1032 can only load generators packaged as part of the Akka-gRPC JARs: https://github.com/akka/akka-grpc/blob/2d1242466e42821b70347ce329c1bd8a1f0db78d/sbt-plugin/src/main/scala/akka/grpc/sbt/AkkaGrpcPlugin.scala#L193-L205.

`akkaGrpcExtraGenerators` is not documented nor tested via a scripted, so maybe it was not meant to be used this way?

The following workaround seems to work:
```scala
      Seq(Compile, Test).flatMap(inConfig(_) {
        Seq(
          PB.artifactResolver := {
            val prev = PB.artifactResolver.value
            val metaBuildClasses =
              (ThisBuild / baseDirectory).value / "project/target/scala-2.12/sbt-1.0/classes/"
            artifact: protocbridge.Artifact => prev(artifact) :+ metaBuildClasses
          },
        )
      })
```

I'll try to open a PR with a failing scripted to demonstrate this.

**Comment by raboof** [comment]
> since #1032 can only load generators packaged as part of the Akka-gRPC JARs

Ah, I didn't realize that (though in retrospect it makes sense). That was not intentional: while indeed we haven't promoted akkaGrpcExtraGenerators much so far, it was definitely intended to allow downstream projects to add additional generators.

**Comment by bjaglin** [comment]
https://github.com/bjaglin/akka-grpc/commits/akkaGrpcExtraGenerators is a repo with 2 commits:
- https://github.com/bjaglin/akka-grpc/commit/4e7bc05e4acb8c4131563bf57009faa3a6c47ce1 describes the usage that works on 1.0.0 in a scripted
- https://github.com/bjaglin/akka-grpc/commit/bf01972796b6607c698cf0f9fe796f7bd1058c8c merges 1.0.1 on top of the previous commit, making the scripted regress (`ava.lang.ClassNotFoundException: generators.CustomGenerator$`)
- https://github.com/bjaglin/akka-grpc/commit/e49553e8045f511e80f31d9624f84122ba6ededb "valides" the workaround documented above

**Comment by bjaglin** [comment]
Now that generators are sandboxed, I guess declaring/compiling custom generators in the metabuild is unnecessarily complex (and unsafe as the sbt classpath is not really under control). They could instead be declared in a normal sub-project, referenced somehow by the build so that the sbt plugin knows where to load `akkaGrpcExtraGenerators` from.

For custom generators published and available on maven/ivy repos, it would also be nice to be able to pass a `ModuleID`, but I guess this is slightly out of the scope of this ticket since it's a completely new feature.

**Comment by raboof** [comment]
> That was not intentional: while indeed we haven't promoted `akkaGrpcExtraGenerators` much so far, it was definitely intended to allow downstream projects to add additional generators

Looking a bit closer, it might be interesting to deprecate `akkaGrpcExtraGenerators` and promote users to add their generators directly to `akkaGrpcGenerators` (or even to `PB.targets`) - that way everyone can decide for themselves whether or not to use sandboxing.

https://github.com/playframework/play-samples/pull/111/files shows what that could look like for `akkaGrpcGenerators` after merging #1115

**Comment by raboof** [comment]
Fixed in #1115
